### PR TITLE
Implement the remove backend for TCP protocol

### PR DESCRIPTION
**Pull request by NotBad4U**
Work for #453 
NOTE: It's under a `TODO` item

### CODE REVIEW

File: lib/src/network/tcp.rs [hunk 1]
```diff
@@ -728,8 +728,17 @@ impl ServerConfiguration {
   }
 
   pub fn remove_backend(&mut self, app_id: &str, backend_address: &SocketAddr, event_loop: &mut Poll) -> Option<ListenToken>{
-      // ToDo
-      None
+    let backend = self.backends.remove_backend(app_id, backend_address);
+
+    let front_token = self.fronts.remove(app_id);
```
Comment (Geal) [inline lib/src/network/tcp.rs]: why are you removing the frontend? There can be multiple backends for one frontend

Comment (NotBad4U) [reply lib/src/network/tcp.rs]: Oh yeah srry, maybe we should remove the frontend when the last backend is remove ?

Comment (Geal) [reply lib/src/network/tcp.rs]: we should remove the frontend when we receive the `RemoveTcpFront` message

Comment (NotBad4U) [reply lib/src/network/tcp.rs]: Okai but we still have to remove the backend in the [ApplicationListener.back_addresses](https://github.com/sozu-proxy/sozu/blob/a9ccb886bad8e2af43cac6a7185ae5ffadba3b6b/lib/src/network/tcp.rs#L592) ?

Comment (Geal) [reply lib/src/network/tcp.rs]: yes

**Pull request merged**